packages/core/src/components/axes/tool-bar.ts

```diff
@@ -0,0 +1,325 @@
+			data-floating-menu-direction="bottom" role="main">
+			<ul class="bx--overflow-menu-options bx--overflow-menu--flip bx--overflow-menu-options--open"
+				tabindex="-1" role="menu" aria-label="Menu" data-floating-menu-direction="bottom"
+				style="left:${this.overflowMenuX}px; top:${this.overflowMenuY}px;">`;
+		defaultHTML += this.getResetZoomMenuItem();
+		defaultHTML += `</ul></div>`;
+
+		return defaultHTML;
+	}
+
+	getResetZoomMenuItem() {
+		const resetZoomOption = Tools.getProperty(
+			this.model.getOptions(),
+			"toolBar",
+			"toolBarMenuItems",
+			"resetZoom"
+		);
+		if (!resetZoomOption.enabled || !this.isZoomBarEnabled) {
+			return "";
+		} else {
+			return `<li
+						class="bx--overflow-menu-options__option">
+						<button class="bx--overflow-menu-options__btn" role="menuitem"  title="Reset"
+							data-floating-menu-primary-focus
+							id="${this.resetZoomMenuItemId}">
+							<div class="bx--overflow-menu-options__option-content">
+								${resetZoomOption.text}
+							</div>
+						</button>
+					</li>`;
+		}
+	}
+
+	handleOverflowMenuEvent() {
+		if (
+			this.overflowMenuOptions
+				.selectAll("ul.bx--overflow-menu-options--open")
+				.size() > 0
+		) {
+			// hide overflow menu
+			this.setOverflowMenuIconHover(false);
+			this.services.events.dispatchEvent(Events.Toolbar.HIDE_MENU_ITEMS);
+		} else {
+			// show overflow menu
+			this.setOverflowMenuIconHover(true);
+			this.services.events.dispatchEvent(Events.Toolbar.SHOW_MENU_ITEMS);
+			const self = this;
+			const resetZoomButtonElement = document.getElementById(
+				this.resetZoomMenuItemId
+			);
+			if (resetZoomButtonElement !== null) {
+				resetZoomButtonElement.addEventListener(
+					"click",
+					function () {
+						self.zoomService.resetZoomDomain();
+
+						self.setOverflowMenuIconHover(false);
+
+						self.services.events.dispatchEvent(
+							Events.Toolbar.HIDE_MENU_ITEMS
+						);
+					},
+					true
+				);
+			}
+		}
+		event.stopImmediatePropagation();
+	}
+
+	setOverflowMenuIconHover(enable) {
+		const overflowIconClass = enable
+			? "icon-overflowRect-hover"
+			: "icon-overflowRect";
+
+		this.overflowMenuIcon.attr("class", overflowIconClass);
+	}
+
+	getZoomInButtonConfig() {
+		return {
+			id: "toolbar-zoomIn",
+			iconId: "zoomInRect",
+			iconSVG: (startPosition) => this.getZoomInIcon(startPosition),
+			clickFunction: () => this.zoomService.zoomIn()
+		};
+	}
+	getZoomOutButtonConfig() {
+		return {
+			id: "toolbar-zoomOut",
+			iconId: "zoomOutRect",
+			iconSVG: (startPosition) => this.getZoomOutIcon(startPosition),
+			clickFunction: () => this.zoomService.zoomOut()
+		};
+	}
+
+	getOverFlowMenuButtonConfig() {
+		return {
+			id: "toolbar-overflow-menu",
+			iconId: "overflowRect",
+			iconSVG: (startPosition) => this.getOverflowMenuIcon(startPosition),
+			clickFunction: () => this.handleOverflowMenuEvent()
+		};
+	}
+
+	getZoomInIcon(startPosition) {
+		return `<?xml version="1.0" encoding="utf-8"?>
```

Comment:
SVGs aren't cleaned up

Reply:
also why does `x` need to be positioned through `startPosition`?

Reply:
We need to set `x` in <svg> and we knows what `x` should be in render(). So we need a parameter for render() to pass the correct `x` for each svg icon. Only render() knows how many icons should be displayed..